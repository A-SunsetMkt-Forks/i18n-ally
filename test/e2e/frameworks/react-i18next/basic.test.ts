import { getExt, openFile, Global, is, not, expect, timeout, setupTest } from '../../ctx'

setupTest('React i18next', () => {
  it('Navigate', async() => {
    await openFile('package.json')
  })

  it('Enabled', () => {
    const ext = getExt()
    is(ext?.isActive, true)
    ext.activate()
  })

  it('Global State', async() => {
    not(Global, undefined)
    is(Global.enabled, true)
    is(Global.enabledFrameworks.length, 2)
    is(Global.enabledFrameworks[0].id, 'react')
    is(Global.enabledFrameworks[1].id, 'i18next')
  })

  it('getCoverage', async() => {
    await timeout(1000)
    not(Global, undefined)
    not(Global.loader, undefined)
    // @ts-ignore
    expect(Global.loader.getCoverage('en')).to.matchSnapshot()
  })
})
