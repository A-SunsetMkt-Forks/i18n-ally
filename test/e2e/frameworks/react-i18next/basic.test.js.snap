// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React i18next getCoverage 1`] = `
Object {
  "emptyKeys": Array [],
  "locale": "en",
  "missing": 0,
  "missingKeys": Array [],
  "total": 6,
  "totalKeys": Array [
    "pages.home.title",
    "translation.description.part1",
    "translation.description.part2",
    "translation.foo.bar",
    "translation.title",
    "translation.titlew",
  ],
  "translated": 6,
  "translatedKeys": Array [
    "pages.home.title",
    "translation.description.part1",
    "translation.description.part2",
    "translation.foo.bar",
    "translation.title",
    "translation.titlew",
  ],
}
`;
